retry.go

```diff
@@ -44,6 +49,15 @@ type requestGUIDReplacer interface {
 	replace() *url.URL
 }
 
+// Get the request ID from the context if specified, otherwise generate one
+func getRequestID(ctx context.Context) string {
+	requestID, ok := ctx.Value(SnowflakeRequestIDKey).(string)
+	if ok && requestID != "" {
+		return requestID
+	}
+	return uuid.New().String()
+}
```

Comment:
Instead of making `SnowflakeRequestIDKey` public, suggest you make it private and add a public function for setting the request ID. Example:

```
func WithRequestID(ctx context.Context, requestID string) context.Context {
  return context.WithValue(ctx, snowflakeRequestIDKey, requestID)
}
```

Usage outside of gosnowflake:

```
ctx = gosnowflake.WithRequestID(context.Background(), "my-request-id")
```